packages/devtools_app/lib/src/provider/provider_screen.dart

```diff
@@ -0,0 +1,115 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:provider/provider.dart' as provider show Provider;
+
+import '../banner_messages.dart';
+import '../screen.dart';
+import '../split.dart';
+import './instance_viewer/instance_details.dart';
+import './instance_viewer/instance_providers.dart';
+import './instance_viewer/instance_viewer.dart';
+import 'provider_list.dart';
+
+final _hasErrorProvider = Provider.autoDispose<bool>((ref) {
+  if (ref.watch(providerIdsProvider) is AsyncError) return true;
+
+  final selectedProviderId = ref.watch(selectedProviderIdProvider).state;
+
+  if (selectedProviderId == null) return false;
+
+  final instance = ref.watch(
+    rawInstanceProvider(InstancePath.fromProviderId(selectedProviderId)),
+  );
+
+  return instance is AsyncError;
+});
+
+class ProviderScreen extends Screen {
+  const ProviderScreen()
+      : super.conditional(
+          id: id,
+          requiresLibrary: 'package:provider/',
+          title: 'Provider',
+          requiresDebugBuild: true,
+          icon: Icons.palette,
+        );
+
+  static const id = 'provider';
+
+  @override
+  Widget build(BuildContext context) {
+    return const ProviderScreenBody();
+  }
+}
+
+class ProviderScreenBody extends ConsumerWidget {
+  const ProviderScreenBody({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context, ScopedReader watch) {
+    final splitAxis = Split.axisFor(context, 0.85);
+
+    // A provider will automatically be selected as soon as one is detected
+    final selectedProviderId = watch(selectedProviderIdProvider).state;
+
+    return ProviderListener<bool>(
+      provider: _hasErrorProvider,
+      onChange: (context, hasError) {
+        if (hasError) showProviderErrorBanner(context);
+      },
+      child: Split(
+        axis: splitAxis,
+        initialFractions: const [0.33, 0.67],
+        children: [
+          const _SplitBorder(child: ProviderList()),
+          Column(
+            children: [
+              if (selectedProviderId != null) ...[
+                Expanded(
+                  child: _SplitBorder(
+                    child: InstanceViewer(
+                      rootPath: InstancePath.fromProviderId(selectedProviderId),
+                    ),
+                  ),
+                ),
+              ] else
+                const Expanded(
+                  child: _SplitBorder(child: SizedBox.expand()),
+                )
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+void showProviderErrorBanner(BuildContext context) {
+  provider.Provider.of<BannerMessagesController>(
+    context,
+    listen: false,
+  ).addMessage(
+    const ProviderUnknownErrorBanner(screenId: ProviderScreen.id)
+        .build(context),
+  );
+}
+
+class _SplitBorder extends StatelessWidget {
```

Comment:
Can you attach an updated screenshot to the CL?
I'm interested in seeing how this widget looks as part of the UI.

Reply:
![Screenshot 2021-04-01 at 03 43 03](https://user-images.githubusercontent.com/20165741/113236212-5eebd780-929c-11eb-8504-ecd68b5ac33c.png)

Reply:
Can you expand out a list and a map in the screenshot and show a screenshot where the hashcodes are fixed or omitted if they are -1.
Hashcodes should also be truncated 5 char hex to match what Flutter does in its debug print.

Reply:
I don't see the split border in the screenshot. I would have expected to see a border that was the focus color.

Reply:
The split border is the thin grey border

But maybe `focusColor` is not the correct variable to use

Reply:
![Screenshot 2021-04-01 at 04 35 19](https://user-images.githubusercontent.com/20165741/113239959-ab86e100-92a3-11eb-9d8a-8272a9747abf.png)